mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,259 @@
+        if (*mask == 'x' && *data != *pattern)
+            return false;
+
+    return (*mask == 0);
+}
+
+//---------------------------------------------------------------------------------------------------------
+// Finds a pattern of bytes in the engine memory given a signature and a mask
+// Returns the address of the first (and hopefully only) match with an optional offset, otherwise nullptr
+//---------------------------------------------------------------------------------------------------------
+void* CEngineBinary::FindPattern(const char* pattern, const char* mask, size_t offset)
+{
+    auto maskLength = strlen(mask);
+    for (size_t i = 0; i <= m_iModuleSize - maskLength; ++i)
+    {
+        auto addr = reinterpret_cast<char*>(m_pModuleBase) + i;
+        if (DataCompare(addr, pattern, mask))
+            return reinterpret_cast<void*>(addr + offset);
+    }
+
+    return nullptr;
+}
+
+bool CEngineBinary::SetMemoryProtection(void* pAddress, size_t iLength, int iProtection)
+{
+#ifdef _WIN32
+    // VirtualProtect requires a valid pointer to store the old protection value
+    DWORD tmp;
+    DWORD prot;
+
+    switch (iProtection)
+    {
+    case MEM_READ:
+        prot = PAGE_READONLY; break;
+    case MEM_READ | MEM_WRITE:
+        prot = PAGE_READWRITE; break;
+    case MEM_READ | MEM_EXEC:
+        prot = PAGE_EXECUTE_READ; break;
+    default:
+    case MEM_READ | MEM_WRITE | MEM_EXEC:
+        prot = PAGE_EXECUTE_READWRITE; break;
+    }
+
+    return VirtualProtect(pAddress, iLength, prot, &tmp);
+#else //POSIX
+    return mprotect(LALIGN(pAddress), iLength + LALDIF(pAddress), iProtection) == 0;
+#endif //_WIN32
+}
+
+void CEngineBinary::ApplyAllPatches()
+{
+#if !defined (OSX) // No OSX patches
+    for (int i = 0; i < sizeof(g_EnginePatches) / sizeof(*g_EnginePatches); i++)
+        g_EnginePatches[i].ApplyPatch();
+#endif
+}
+
+CEngineBinary g_EngineBinary;
+
+void CEnginePatch::ApplyPatch()
+{
+    if (!m_pPatch)
+    {
+        Warning("Engine patch \"%s\" FAILED: No value provided\n", m_sName);
+        return;
+    }
+
+    void* addr = CEngineBinary::FindPattern(m_pSignature, m_pMask, m_iOffset);
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *reinterpret_cast<uintptr_t**>(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+        if (CEngineBinary::SetMemoryProtection(pMemory, m_iLength, MEM_READ|MEM_WRITE|MEM_EXEC))
+        {
+            Q_memcpy(pMemory, m_pPatch, m_iLength);
+
+            CEngineBinary::SetMemoryProtection(pMemory, m_iLength, MEM_READ|MEM_EXEC);
+
+            DevLog("Engine patch \"%s\" applied successfully\n", m_sName);
+            return;
+        }
+        Warning("Engine patch \"%s\" FAILED: Could not override memory protection\n", m_sName);
+        return;
+    }
+    Warning("Engine patch \"%s\" FAILED: Could not find signature\n", m_sName);
+    return;
```

Comment:
No need to return on the last line in the function, why not just put this in an else block? Same goes for the other warning